tests/http/clients/starlite.py

```diff
@@ -129,3 +159,82 @@ async def post(
             data=response.content,
             headers=response.headers,
         )
+
+    @contextlib.asynccontextmanager
+    async def ws_connect(
+        self,
+        url: str,
+        *,
+        protocols: List[str],
+    ) -> AsyncGenerator[WebSocketClient, None]:
+        try:
+            with self.client.websocket_connect(url, protocols) as ws:
+                yield StarliteWebSocketClient(ws)
+        except WebSocketDisconnect as error:
+            ws = StarliteWebSocketClient(None)
+            ws.handle_disconnect(error)
+            yield ws
+
+
+class StarliteWebSocketClient(WebSocketClient):
+    def __init__(self, ws: Any):
```

Comment:
nit: AFAIK `ws` would be a `WebSocketTestSession` imported via `from startlite.testing import WebSocketTestSession`